workbench-core/accounts/src/utilities/hostingAccountLifecycleService.ts

```diff
@@ -385,7 +396,13 @@ export default class HostingAccountLifecycleService {
         return output.OutputKey === 'EncryptionKeyArn';
       })!.OutputValue;
 
-      if (removeCommentsAndSpaces(actualTemplate) === removeCommentsAndSpaces(expectedTemplate)) {
+      if (
+        expectedTemplates
```

Comment:
lets pull this out into its own value so its easier to read.

Reply:
+1, also need some comments I think. 

How is this piece of code determining which template to match against? Let's say there has been a big change in all 3 expected templates, then how would this work?

Reply:
Ah yes, I'll add comments and break the code out to make it easier to understand. The logic is the following. If the cfn template for an account matches one of the templates in S3 we'll consider the account is up to date. If a big change was made to all 3 expected templates, then the CFN template won't match any of the 3 expected templates and the account's status will change to `NEEDS_UPDATE`.


This keep the logic simple. However users need to be careful that when they manually onboard a hosting account, they choose the right CFN template.